gum/backend-linux/gumprocess-linux.c

```diff
@@ -869,6 +876,102 @@ gum_emit_executable_module (const GumModuleDetails * details,
   return FALSE;
 }
 
+/*
+ * Loading an executable by passing it as an argument to a loader is often used
+ * to run 32-bit binaries on 64-bit kernels, e.g.
+ *
+ * /usr/arm-linux-gnueabi/lib/ld-2.27.so ./myexe
+ *
+ * We detect this scenario by the absence of a '.interp' section in the binary
+ * refrenced by /proc/self/exe. If this is the case, then we use
+ * /proc/self/cmdline to determine the process command line and extract argv[1]
+ * to determine the name of the main executable which the loader was used to
+ * load.
+ *
+ * We then search the list of modules from /proc/self/maps to find a match for
+ * the name using the basename from both argv[1] and the entry in the map. We
+ * then emit the module described by this name as the main application
+ * executable.
+ */
+static gboolean
+gum_handle_interpreter (const GumModuleDetails * details,
+                        gpointer user_data)
+{
+  GumEmitExecutableModuleContext * ctx = user_data;
+  GumElfModule * module;
+  gboolean has_interp;
+  gchar * contents;
+  gsize length;
+  gsize i;
+  gboolean result = TRUE;
```

Comment:
Should move up to right after `ctx` as per https://github.com/frida/frida/blob/master/CONTRIBUTING.md#rule-21 exception 1.

Reply:
Let's name it `handled` for clarity.